citesphere/src/main/java/edu/asu/diging/citesphere/web/user/ShowPeopleController.java

```diff
@@ -0,0 +1,125 @@
+    public String showPeople(Model model, Authentication authentication, @PathVariable("zoteroGroupId") String groupId,
+            @RequestParam(defaultValue = "1", required = false, value = "page") String page,
+            @RequestParam(defaultValue = "title", required = false, value = "sort") String sort) {
+        Integer pageInt = 1;
+        try {
+            pageInt = new Integer(page);
+        } catch (NumberFormatException ex) {
+            logger.warn("Trying to access invalid page number: " + page);
+        }
+        IUser user = (IUser) authentication.getPrincipal();
+        CitationResults results;
+        try {
+            results = citationManager.getGroupItems(user, groupId, null, pageInt, sort);
+        } catch (ZoteroHttpStatusException e) {
+            logger.error("Exception occured", e);
+            return "error/500";
+        } catch (GroupDoesNotExistException e) {
+            logger.error("Exception occured", e);
+            return "error/404";
+        }
+        model.addAttribute("zoteroGroupId", groupId);
+        model.addAttribute("currentPage", pageInt);
+        model.addAttribute("sort", sort);
+
+        Map<IPerson, List<ICitation>> personCitationMap = new HashMap<IPerson, List<ICitation>>();
+
+        results.getCitations().stream()
+                .forEach(
+                        iCitation -> (Stream
+                                .of(iCitation.getAuthors().stream(), iCitation.getEditors().stream(),
+                                        iCitation.getOtherCreators().stream()
+                                                .map(iOtherCreator -> iOtherCreator.getPerson()))
+                                .reduce(Stream::concat).orElseGet(Stream::empty))
+                                        .filter(distinctByKey(p -> p.getFirstName())).forEach(au -> {
+                                            boolean containKey = false;
+                                            for (IPerson key : personCitationMap.keySet()) {
+                                                if (key.getFirstName().equals(au.getFirstName())
+                                                        && key.getLastName().equals(au.getLastName())) {
+                                                    personCitationMap.get(key).add(iCitation);
+                                                    containKey = true;
+                                                    break;
+                                                }
+                                            }
+                                            if (!containKey) {
+                                                List<ICitation> citList = new ArrayList<ICitation>();
+                                                citList.add(iCitation);
+                                                personCitationMap.put(au, citList);
+                                            }
+
+                                        }));
```

Comment:
all the different nested streams here make it very hard to follow. This is a case were mixing streams and regular loops would be preferable.